### PR TITLE
Handle error for audio playback.

### LINKED ISSUE #44: Handle error when creating AudioTrack and writing data
Creating an AudioTrack is not guaranteed to be successful. Writing data can also fail due to resource is changed on device. In that case, the error should be handled.

### PR DESCRIPTION
**Pull request by flamme**
Fixes #44.

### CODE REVIEW

**Review by robertwu1 — APPROVED** [review]

File: app/src/main/java/com/google/android/soundchecker/utils/AudioTrackSink.kt [hunk 1]
```diff
@@ -55,9 +56,14 @@ class AudioTrackSink(var preferredDevice: AudioDeviceInfo? = null,
             mFloatBuffer = FloatArray(getChannelCount() * FRAMES_PER_BURST)
         }
         if (mAudioTrack == null) {
-            mAudioTrack = createAudioTrack(0)
+            try {
+                mAudioTrack = createAudioTrack(0)
+            } catch (_: Exception) {
+                errorCallback?.onError(AudioErrorCallback.ERROR, "Failed to create AudioTrack")
```
Comment (philburk) [inline app/src/main/java/com/google/android/soundchecker/utils/AudioTrackSink.kt:62]: Can we use getMessage() on the Exception to get useful information.

File: app/src/main/java/com/google/android/soundchecker/utils/AudioTrackSink.kt [hunk 2]
```diff
@@ -66,7 +72,9 @@ class AudioTrackSink(var preferredDevice: AudioDeviceInfo? = null,
     }
 
     override fun start() {
-        open()
+        if (open() != AudioErrorCallback.SUCCESS) {
+            return
```
Comment (philburk) [inline app/src/main/java/com/google/android/soundchecker/utils/AudioTrackSink.kt:76]: Are we failing silently?

File: app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt [hunk 3]
```diff
@@ -318,4 +323,13 @@ open class BaseFilePlayerActivity : ComponentActivity(), OnAudioFocusChangeListe
             }
         }
     }
+
+    inner class FilePlaybackErrorCallback : AudioErrorCallback() {
+        override fun onError(error: Int, msg: String) {
+            if (error != AudioErrorCallback.SUCCESS) {
+                Log.e(mTag, "Stop playback when receiving error=$error, msg=$msg")
```
Comment (philburk) [inline app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt:330]: Will the user see this error?
When writing framework code, logging the error and returning an error ode is enough.
But when writing test apps, if we get an unexpected error from a bug, then we should alert the user.

**Review by philburk — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt (hunk 3 above)
Comment (flamme) [reply app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt:336]: Added a text to show error message

File: app/src/main/java/com/google/android/soundchecker/utils/AudioTrackSink.kt (hunk 2 above)
Comment (flamme) [reply app/src/main/java/com/google/android/soundchecker/utils/AudioTrackSink.kt:77]: When it fails, the caller knows via error callback.

**Review by philburk — APPROVED** [review]

**Pull request merged**